backend/api/src/test/kotlin/software/shonk/adapters/incoming/ShorkInterpreterControllerV1IT.kt

```diff
@@ -104,4 +104,111 @@ class ShorkInterpreterControllerV1IT() : AbstractControllerTest() {
         assertEquals("FINISHED", responseData["gameState"])
         assertEquals("B", responseData["result.winner"])
     }
+
+    @Test
+    fun `test player code submission with valid username`() = runTest {
+        val player = "playerA"
+        client.post("/api/v1/lobby/0/code/$player") {
+            contentType(ContentType.Application.Json)
+            setBody("someString")
+        }
+        val result = client.get("/api/v1/lobby/0/code/playerA")
+        assertEquals(
+            "someString",
+            Json.parseToJsonElement(result.bodyAsText()).jsonObject["code"]?.jsonPrimitive?.content,
+        )
+    }
```

Comment:
This scenario is basically already covered in the test `testGetPlayerCode` in this same file.
You could rename that testcase to show that it tests both `POST` and `GET` and change it to use the v1 endpoints, instead of the v0 endpoints (which needs to happen anyways)

Reply:
I changed the name so that it could fit both POST and GET endpoints

Reply:
yes, but the reason I specifically mentioned the other test is because 
- both tests would now do the same thing; either one should thus be deleted
- the pre-existing test tested with two players uploading, which I would prefer to see here

so in short:
I'd like to see only one test testing this behaviour and I would like the test to upload for two players

Reply:
I  understood what you mean now, then I guess your recommendation is to remove this test since they are very similar? I would remove my test then and keep the older one, after implementation we can add the new endpoint to the code to replace the apiv0

Reply:
pretty much, except it'd be good if you already changed the test to use v1 in this branch, so we know that the code is working before merging.

Reply:
alright I had replaced the apiv0 POST with the v1 one and the tests are working well